content/profiling_interface.tex

```diff
@@ -0,0 +1,196 @@
+The objective of the \openshmem profiling interface is to ensure an 
+easy and flexible usage model for profiling (and other similar) 
+tool developers to interface their codes into \openshmem 
+implementations on different platforms. Since \openshmem is a 
+machine-independent standard with different implementations, it is 
+unreasonable to expect that the authors and developers of profiling 
+tools for \openshmem will have access to the source code that 
+implements \openshmem on any particular machine. It is, therefore,  
+necessary to provide a mechanism by which the implementors of such 
+tools can collect whatever performance information they wish 
+\textit{without} access to the underlying implementation.
+
+The \openshmem profiling interface places the following requirements 
+on implementations. 
+
+\begin{enumerate}
+\item An \openshmem implementation must provide a mechanism through 
+which all of the \openshmem defined functions may be accessible 
+with a name shift. This requires an alternate 
+entry point name, with the prefix \FUNC{pshmem\_} for each 
+\openshmem function. For \openshmem inlined functions (e.g. macros), 
+it is also required that the \FUNC{pshmem\_} version is supplied 
+although it is not possible to replace the \FUNC{pshmem\_} version 
+with a user-defined version at link time.
+\item It must be ensured that the \openshmem functions that are not 
+replaced as above, may still be linked into an executable image 
+without causing name clashes. 
+\item Documentation of the implementation of different language 
+bindings of the \openshmem interface must indicate if they 
+are layered on top of each other. Using this documentation, a  
+developer can determine whether he/she needs to implement the 
+profile interface for each binding or not. For example, it must 
+be noted that the \openshmem \Cstd[11] type-generic interfaces for 
+different RMA and AMO operations cannot have any equivalent 
+\FUNC{pshmem\_} interfaces because of the limitation from \Cstd 
+\FUNC{\_Generic} semantics. 
```

Comment:
Suggest: "For example, it must be noted that the \openshmem \Cstd[11] type-generic interfaces for 
different RMA and AMO operations cannot have any equivalent \FUNC{pshmem\_} interfaces because the \Cstd[11] type-generic interfaces are implemented as macros."